### ISSUE TITLE
Solve fast typing enter key sync error

### ISSUE THREAD

**Issue by neitanod** [ctrlpvim/ctrlp.vim#284]
TL;DR: What I'm asking for:  Process the <CR> key as part of the input queue (just like any other key) instead of discarding all previous input that the CPU could not process yet and instantly processing the <CR> key.

As I see it, there is an error in the way the keys are accepted by the plugin.  For example:

Let's say I have a large project with lots of files in it.  Among those files there are two files called  'index.php' and 'form.php' inside a path '/web/setup/':

/web/setup/index.php
/web/setup/form.php  

If I type  "<C-p>setup" and wait a few hundred milliseconds those two matches are presented as first options.
If I were to open "form.php" I would just type <CR> and that would be Ok, but what I want to open is "index.php", so I need to type something like "ind", to get ctrlp to focus on "index.php".

The problem is that you have to wait about a second between typing "ind" and <CR>, otherwise ctrlp would simply discard the new charactes and it would just open the file it's focused on.  It would never complete the search, and it would be opening the wrong file.

It's not the speed or the delay I'm complaining about, but the fact that when the <CR> key is pressed it's instantly accounted for and all prevoiusly typed characters that were not yet processed get discarded and the user search gets truncated.  It even gets truncated differently in different speed CPUs.

Seems subtle, but this problem prevents the user -specially the fast-typing ones- from mindlessly typing the search term followed by <CR>, and for me that beats the purpose of the plugin, which is to "follow the Vim way", that I jokingly define as "editing with your eyes closed", but could be defined in a more serious way as "always prefer the commands that do not require feedback over those that require some feedback".

**Comment by mattn** [comment]
Do you set `let g:ctrlp_lazy_update = 1` in your vimrc?
